src/main/java/net/mcreator/ui/modgui/ModElementGUI.java

```diff
@@ -425,56 +425,67 @@ protected final void finalizeGUI(boolean wrapInScrollPane) {
 	}
 
 	private void disableUnsupportedFields() {
-		List<String> inclusions = mcreator.getGeneratorConfiguration()
-				.getSupportedDefinitionFields(modElement.getType());
-
 		List<String> exclusions = mcreator.getGeneratorConfiguration()
 				.getUnsupportedDefinitionFields(modElement.getType());
 
-		if (inclusions != null && exclusions != null) {
-			LOG.warn("Field inclusions and exclusions can not be used at the same time. Skipping them.");
-			return;
-		}
+		List<String> inclusions = mcreator.getGeneratorConfiguration()
+				.getSupportedDefinitionFields(modElement.getType());
 
-		if (inclusions != null) {
-			Field[] fields = getClass().getDeclaredFields();
-			for (Field field : fields) {
-				if (Component.class.isAssignableFrom(field.getType())) {
-					if (!inclusions.contains(field.getName())) {
-						try {
-							field.setAccessible(true);
-							Component obj = (Component) field.get(this);
+		if (exclusions != null && inclusions != null) { // can't exclude and include together
+			LOG.warn("Field exclusions and inclusions can not be used at the same time. Skipping them.");
+		} else if (exclusions != null && !exclusions.isEmpty() || inclusions != null && !inclusions.isEmpty()) {
+			Map<Container, List<Component>> unsupportedComps = new HashMap<>();
```

Comment:
This map actually contains components supported, so wouldn't `supportedComponents` be a better name?

Reply:
>Maybe.